CMakeLists.txt

```diff
@@ -3,20 +3,49 @@ cmake_minimum_required(VERSION 2.6.4)
 # http://www.cmake.org/Bug/view.php?id=8598
 # http://gcc.gnu.org/bugzilla/show_bug.cgi?id=35707
 # http://www.cmake.org/Bug/view.php?id=8408
-project(PARTMC Fortran C)
+if (ENABLE_TCHEM)
+  # Use the wrapper to nvcc for the CXX compiler for GPU
+  if (ENABLE_TCHEM_GPU)
+    set(ENV{CXX} $ENV{KOKKOS_PATH}/bin/nvcc_wrapper)
+    project(PARTMC Fortran C CXX)
+  else()
+    project(PARTMC Fortran C)
+  endif()
+else()
+  project(PARTMC Fortran C)
+endif()
 
 set(PACKAGE_BUGREPORT "mwest@illinois.edu")
 set(PACKAGE_NAME "PartMC")
 set(PACKAGE_STRING "PartMC 2.8.0")
 set(PACKAGE_TARNAME "partmc")
 set(PACKAGE_VERSION "2.8.0")
 
+include(FortranCInterface)
+
+# Following must be commented out if using -Wimplicit-interface with -Werror
+# Working with C and Fortran
+#FortranCInterface_VERIFY()
+#if (ENABLE_TCHEM_GPU)
+#   # Working with C++ and Fortran
+#   FortranCInterface_VERIFY(CXX)
+#endif()
+
+if (ENABLE_TCHEM)
+  set(CMAKE_C_FLAGS "-lpthread -ldl -fopenmp" CACHE STRING "" FORCE)
+  set(CMAKE_Fortran_FLAGS "-lpthread -ldl -fopenmp" CACHE STRING "" FORCE)
+  if (ENABLE_TCHEM_GPU)
+    set(CMAKE_CXX_FLAGS "-lpthread -ldl -fopenmp" CACHE STRING "" FORCE)
+  endif()
+endif()
```

Comment:
I don't understand enough about the order in which CMake initializes/reads variables like `CMAKE_C_FLAGS`. I know that this can be customized by the user later. Is this above setting the right way to do it?

Reply:
I think I've resolved all this so that we aren't needlessly aggressive with the settings. The user can now customize their compiler flags without us having to interfere or in the orignal case, actually force the settings. Fixed in https://github.com/compdyn/partmc/pull/198/commits/986c8a2a606cee5f9927d30226f498f78e6a1edc by moving libraries and https://github.com/compdyn/partmc/pull/198/commits/3463e7d6036620ce1d0b66e61fbac837e3c08bd3 for OpenMP.